src/ui/control/logo_control.js

```diff
@@ -38,15 +45,12 @@ class LogoControl {
 
     _updateLogo(e) {
         if (e && e.sourceDataType === 'metadata') {
-            if (!this._container.childNodes.length && this._logoRequired()) {
-                const anchor = DOM.create('a', 'mapboxgl-ctrl-logo');
-                anchor.target = "_blank";
-                anchor.href = "https://www.mapbox.com/";
-                anchor.setAttribute("aria-label", "Mapbox logo");
-                this._container.appendChild(anchor);
-                this._map.off('data', this._updateLogo);
-            } else if (this._container.childNodes.length && !this._logoRequired()) {
-                this.onRemove();
+            if (!this._logoDisplay && this._logoRequired()) {
+                this._container.style.display = 'block';
+                this._logoDisplay = true;
+            } else if (this._logoDisplay && !this._logoRequired()) {
+                this._container.style.display = 'none';
+                this._logoDisplay = false;
```

Comment:
We could simply toggle display like this:
```
this._container.style.display = this._logoRequired() ? 'block' : 'none'
```

The state variable `this._logoDisplay` seems unnecessary.